pygen/pygen_src/riscv_instr_cover_group.py

```diff
@@ -0,0 +1,633 @@
+        def __init__(self, instr):
+            super().__init__(instr)
+
+            self.cp_imm_sign = vsc.coverpoint(lambda: instr.imm_sign,
+                                              cp_t=vsc.enum_t(operand_sign_e))
+            self.cp_sign_cross = vsc.cross([self.cp_rs1_sign,
+                                            self.cp_imm_sign])
+
+    @vsc.covergroup
+    class sltu_cg(cmp_instr_cg):
+        def __init__(self, instr):
+            super().__init__(instr)
+
+            self.cp_rs2 = vsc.coverpoint(lambda: instr.rs2,
+                                         cp_t=vsc.enum_t(riscv_reg_t))
+            self.cp_rs2_sign = vsc.coverpoint(lambda: instr.rs2_sign,
+                                              cp_t=vsc.enum_t(operand_sign_e))
+            self.cp_sign_cross = vsc.cross([self.cp_rs1_sign,
+                                            self.cp_rs2_sign])
+
+    @vsc.covergroup
+    class sltiu_cg(cmp_instr_cg):
+        def __init__(self, instr):
+            super().__init__(instr)
+
+            self.cp_imm_sign = vsc.coverpoint(lambda: instr.imm_sign,
+                                              cp_t=vsc.enum_t(operand_sign_e))
+            self.cp_sign_cross = vsc.cross([self.cp_rs1_sign,
+                                            self.cp_imm_sign])
+
+    '''Branch instructions'''
+
+    @vsc.covergroup
+    class beq_cg(sb_instr_cg):
+        def __init__(self, instr):
+            super().__init__(instr)
+
+    @vsc.covergroup
+    class bne_cg(sb_instr_cg):
+        def __init__(self, instr):
+            super().__init__(instr)
+
+    @vsc.covergroup
+    class blt_cg(sb_instr_cg):
+        def __init__(self, instr):
+            super().__init__(instr)
+
+    @vsc.covergroup
+    class bge_cg(sb_instr_cg):
+        def __init__(self, instr):
+            super().__init__(instr)
+
+    @vsc.covergroup
+    class bltu_cg(sb_instr_cg):
+        def __init__(self, instr):
+            super().__init__(instr)
+
+    @vsc.covergroup
+    class bgeu_cg(sb_instr_cg):
+        def __init__(self, instr):
+            super().__init__(instr)
+
+    '''Jump instructions'''
+
+    @vsc.covergroup
+    class jal_cg(j_instr_cg):
+        def __init__(self, instr):
+            super().__init__(instr)
+
+            self.cp_imm_align = vsc.coverpoint(lambda: instr.imm[1],
+                                               bins={
+                                                   "Aligned"    : vsc.bin(1),
+                                                   "Not-aligned": vsc.bin(0)
+                                               }
+                                               )
+
+    @vsc.covergroup
+    class jalr_cg(j_instr_cg):
+        def __init__(self, instr):
+            super().__init__(instr)
+
+            '''default bins are not supported in pyvsc. We ignore it here 
+            as coverage values hit in default bin are not taken account while 
+            reporting coverage'''

```

Comment:
is this a known issue in PyVSC?

Reply:
Yeah it's actually one of the features that is supported on the SV side but is not supported in pyvsc, I will ask them to see if they are going to support it in the near future.

Reply:
sounds good!